collect/cache/cuckoo.go

```diff
@@ -0,0 +1,76 @@
+package cache
+
+import (
+	"sync"
+
+	cuckoo "github.com/panmari/cuckoofilter"
+)
+
+// This wraps a cuckoo filter implementation in a way that lets us keep it running forever
+// without filling up.
+// You must call Maintain() periodically, most likely from a goroutine.
+// We maintain two filters that are out of sync; when current is full, future is half full
+// and we then discard current and replace it with the future, and then create a new, empty future.
+// This means that we need to use the firstTime flag to avoid putting anything into future
+// until after current reaches .5.
+type CuckooTraceChecker struct {
+	current   *cuckoo.Filter
+	future    *cuckoo.Filter
+	mut       sync.RWMutex
+	capacity  uint
+	firstTime bool
```

Comment:
nit: I'd prefer a name like like `halfEmpty` or `insertIntoBoth` (that one may change some of the if statements).  I'm still struggling with `firstTime` because it feels like it is tracking something that only happens 1 time, but it reality `c.current.Insert([]byte(traceID))` could happen many times before `firstTime` is set to false.

Reply:
I've reworked this to eliminate this entirely and just leave future as nil until it's time to use it. Also has a lot more comments.